### DISCUSSION TITLE
Authenticate to Azure FHIR API (azurehealthcareapis)

### DISCUSSION THREAD

**Discussion by portegah85** [FirelyTeam/firely-net-sdk#2698] [Q&A]
Hi all, Im using  Health Data Services, I need to create the FhirClient calling the Azure FHIR server 

var _fhirServer = 'https://myserver.fhir.azurehealthcareapis.com'
var fhirClient = new FhirClient(_fhirServer)

var resourceUrl = $"Patient/{Id}";    
var patient = fhirClient.Read<Patient>(resourceUrl); // Response: 401 Unauthenticated

I have been playing with the constructor to see if there is any way to pass the bearer token in the call to be able to authenticate successfully

Any help will be great, Thanks!

**Comment by brianpos** [comment]
There's a chat discussion that covers this waaay back.
https://chat.fhir.org/#narrow/stream/179171-dotnet/topic/HttpClient.20based.20FhirClient
You can use a message handler of your own to inject the authorization header to the httpclient passed into the constructor of the FhirClient.

**Comment by mmsmits** [comment]
There is some documentation how to add a custom MessageHandler to the FHIR Client. Here:

https://docs.fire.ly/projects/Firely-NET-SDK/en/latest/client/request-response.html

In the example a basic AuthenticationHandler is used. Hope that helps.
